libs/src/services/discoveryProvider/DiscoveryProvider.ts

```diff
@@ -1465,6 +1470,13 @@ export class DiscoveryProvider {
       })) ?? response
 
     const responseBody: DiscoveryResponse<Response> = await response.json()
+
+    if (blockNumber && responseBody.latest_indexed_block < blockNumber) {
```

Comment:
this type of < blockNumber check should eventually not be needed with a fully shipped storage v2